### DISCUSSION TITLE
Middlware not working directly load url after nuxt generate

### DISCUSSION THREAD

**Discussion by prabodhana** [nuxt/nuxt#9508] [Questions (Nuxt 2)]
hello,

I'm using "nuxt auth module" to authenticate the user and auth module middlewares use to protect routes. The development environment it's working without issue but if I use guest middleware it not redirecting me to the home page if I directly type and go to the URL. After searching on the internet for that issue I found a solution using 'cookie-universal-nuxt'. It works on the development environment what I expected but after generating static it's not working. is there any solution for this?
Thanks.

**Comment by TheAlexLichter** [comment]
Be aware that middleware is executed always server-side on the initial request (and client-side on subsequent requests). This means, it has been triggered during the build phase and will be triggered client-side when navigating from e.g. /no-middleware to /middleware.

**Reply by prabodhana** [reply]
thank for the reply can explain me more I cant clearly understand that
